python/cugraph/tests/dask/mg_context.py

```diff
@@ -54,11 +54,12 @@ class MGContext:
         Number of devices to use, verification must be done prior to call
         to ensure that there are enough devices available.
     """
-    def __init__(self, number_of_devices=None, rmm_managed_memory=False):
+    def __init__(self, number_of_devices=None, rmm_managed_memory=False, cluster=None, p2p=False):
```

Comment:
If a cluster is provided, and ```MGContext``` is used as a context manager the ```__exit__``` section will close the cluster, which might not be the expected behavior (as we might want to reuse the cluster with a different client). Having an extra boolean ```close_on_exit = False``` could give more control over it?

Reply:
I ended up removing the ability to override the entire cluster in the latest commit, since I realized it was already creating a `LocalCUDACluster`. I can add the ability to override the cluster completely (and apply your suggestion to add a flag to not close on exit) in the future if we end up needing that.  The `p2p` arg to the Comms init call to initialize the UCX endpoints was what we really needed for the current tests.